### DISCUSSION TITLE
Converting InputStream to String seem to Corrupt Data

### DISCUSSION THREAD

**Discussion by Homersbm** [dcm4che/dcm4che#943] [Q&A] [answered]
I have a multipart/related; type="application/dicom" response.. so basically its a Dicom content.
The response is in InputStream format.. I am trying to convert it into String as I wanted to remove some extra Line and content.
But the converted String data is (without any data manipulation) corrupted because when i convert it back to dcm File, i see some extra characters.. 

**Can someone please suggest a good way to convert dicom data from Inputstream to String? Am I applying Wrong Encoding?**

```
    private static void write(InputStream in, String fileName) throws IOException {
        Path path = outDir != null
                ? Files.createDirectories(Paths.get(outDir)).resolve(fileName)
                : Paths.get(fileName);
         String result = IOUtils.toString(in, StandardCharsets.UTF_8);
        String ddd = result.substring(result.indexOf("\r\n\r\n") + 4);                                     //Remove Some lines
        InputStream stream = IOUtils.toInputStream(ddd,StandardCharsets.UTF_8);
        try (OutputStream out = Files.newOutputStream(path)) {
            StreamUtils.copy(stream, out);
        }
    }

```

**Comment by gunterze** [comment]
You can't convert a DICOM binary encoded stream into a String! You may use [`class MultipartParser`](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-mime/src/main/java/org/dcm4che3/mime/MultipartParser.java) to unpack the parts from the multipart container. See source of utilities [`wadors`](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-wadors) or [`stowrsd`](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-stowrsd) for how to use it.

**Reply by Homersbm** [reply]
Thanks. I converted the stream into ByteArray in order to remove some lines and it worked.

**Comment by Homersbm** [comment, marked as answer]
I converted the stream into ByteArray in order to remove some lines and it worked.
